Correct docstrings and a typo in comments

lewis-yeung opened the pull request:
## What does this PR do?

Correct docstrings of the following methods:

- `TrainingArguments.set_save`
- `TrainingArguments.set_logging`

## Who can review?

@stevhliu

stevhliu commented on src/transformers/training_args.py:
```diff
@@ -2276,7 +2276,7 @@ def set_save(
     ):
         """
-        A method that regroups all arguments linked to the evaluation.
+        A method that regroups all arguments linked to the checkpoint saving.
```
```suggestion
        A method that regroups all arguments linked to checkpoint saving.
```

stevhliu commented on src/transformers/training_args.py:
```diff
@@ -2329,7 +2329,7 @@ def set_logging(
     ):
         """
-        A method that regroups all arguments linked to the evaluation.
+        A method that regroups all arguments linked to the logging.
```
```suggestion
        A method that regroups all arguments linked to logging.
```

stevhliu reviewed (approved):
Good catch, thanks for the fixes!

lewis-yeung commented on src/transformers/training_args.py:
@stevhliu Should we change this (remove `the` from `the evaluation`) as well?

https://github.com/huggingface/transformers/blob/6cbc1369a330860c128a1ba365f246751382c9e5/src/transformers/training_args.py#L2181

stevhliu replied:
Sure, that'll keep it nice and consistent :)

lewis-yeung replied:
Updated. Also fixed a typo (`sur` -> `sure`) in `trainer.py`:

https://github.com/huggingface/transformers/blob/a2f55a65cd0eb3bde0db4d5102a824ec96c7d7e9/src/transformers/trainer.py#L1904

lewis-yeung committed:
docs(training_args): correct docstrings

Correct docstrings of these methods in `TrainingArguments`:

- `set_save`
- `set_logging`

lewis-yeung committed:
docs(training_args): adjust words in docstrings

Co-authored-by: Steven Liu <59462357+stevhliu@users.noreply.github.com>

lewis-yeung committed:
docs(trainer): correct a typo in comments

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_27047). All of your documentation changes will be reflected on that endpoint.
